src/components/CreateInstructorPage/index.jsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Helmet from 'react-helmet';
+
+import { CreateInstructorForm } from './CreateInstructorForm';
+import StatusAlert from '../StatusAlert';
+import LoadingSpinner from '../LoadingSpinner';
+
+class CreateInstructorPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startedFetching: false,
+    };
+    this.handleInstructorCreate = this.handleInstructorCreate.bind(this);
+    this.setStartedFetching = this.setStartedFetching.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.fetchInstructorOptions();
+    this.setStartedFetching();
+  }
+
+  setStartedFetching() {
+    this.setState({ startedFetching: true });
+  }
+
+  processSocialLinks(links) {
+    if (!links) {
+      return [];
+    }
+
+    return links.map(link => (
+      ({
+        id: null,
+        type: link.type,
+        title: link.title,
+        url: link.url,
+      })
+    ));
+  }
+
+  processAreasOfExpertise(areasOfExpertise) {
+    if (!areasOfExpertise) {
+      return [];
+    }
+
+    return areasOfExpertise.map(area => ({ id: null, value: area.value }));
+  }
+
+  processPosition(title, organization) {
+    return {
+      title,
+      organization,
+      organization_override: null,
+    };
+  }
+
+  handleInstructorCreate(options) {
+    const instructorData = {
+      profile_image: options.image,
+      given_name: options.firstName,
+      family_name: options.lastName,
+      bio: options.bio,
+      major_works: options.majorWorks,
+      areas_of_expertise: this.processAreasOfExpertise(options.areasOfExpertise),
+      urls_detailed: this.processSocialLinks(options.socialLinks),
+      position: this.processPosition(options.title, options.organization),
+    };
+
+    this.props.createInstructor(instructorData);
+  }
+
+  render() {
+    const {
+      instructorOptions,
+      instructorInfo,
+    } = this.props;
+
+    if (!instructorOptions || !instructorInfo) {
```

Comment:
Are these programmer error states or possibly because of some whack network traffic? It's fine either way, just trying to wrap my head around possibilities.

Reply:
Should be network errors